src/core/engine/processors/mongo_order_status_history_distribution.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABCMeta
+
+from src.constants.etl_status import Service
+
+from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.lib.repositories.impl_no_sql.order_status_history_repository_impl import (
+    OrderStatusHistoryRepositoryImpl as MongoOrderStatusHistoryRepository,
+)
+
+
+class MongoOrderStatusHistoryDistribution(AbstractProcessor, metaclass=ABCMeta):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+        self.mongo_order_status_history_repository = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_order_status_history_repository = ioc.get(
+            MongoOrderStatusHistoryRepository
```

Comment:
Any reason why we are not setting this in the constructor?

Reply:
You should follow the same pattern the other processors are following

Reply:
This is the way how it is implemented in others processors.